### DISCUSSION TITLE
Błąd podczas kompilacji

### DISCUSSION THREAD

**Discussion by magic7003** [colobot/colobot#1442] [Q&A]
Witam !

   Mam problem podczas kompilacji Colobota .

   Wynik z kompilacji wygląda następująco :

konrad@konrad-ThinkPad-L:/media/konrad/System/Colobot$ make
[  1%] Built target localename
[ 17%] Built target CBot
[ 64%] Built target colobotbase
[ 64%] Linking CXX executable ../colobot
/usr/bin/ld: libcolobotbase.a(app.cpp.o): undefined reference to symbol 'SDL_JoystickNumAxes'
//usr/lib/i386-linux-gnu/libSDL2-2.0.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/CMakeFiles/colobot.dir/build.make:110: recipe for target 'colobot' failed
make[2]: *** [colobot] Error 1
CMakeFiles/Makefile2:149: recipe for target 'src/CMakeFiles/colobot.dir/all' failed
make[1]: *** [src/CMakeFiles/colobot.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
konrad@konrad-ThinkPad-L:/media/konrad/System/Colobot$ 

   Myślę że zaistalowałem wszystkie biblioteki SDL i SDL2 .

   Proszę o pomoc .

**Comment by ThePolishPPK** [comment]
Jeżeli dobrze rozumiem to biblioteka SDL2 nie została załadowana, a jej powód to:
```
/usr/lib/i386-linux-gnu/libSDL2-2.0.so.0: error adding symbols: DSO missing from command line
```
Na StackOverflow jest dyskusja na temat "DSO missing from command line" [LINK](https://stackoverflow.com/questions/24989432/linking-error-dso-missing-from-command-line)
Przyczyną problemu mogą być niespójne zależności w systemie.
Czyli np. SDL2 wymaga pthread a system twierdzi, że nie, więc **ld**(linker) go nie ładuje.

Jeżeli bardzo ci zależy, aby ręcznie skompilować program, powinieneś ręcznie sprawdzić zależności dla SDL2. Możesz również wirtualizować innego linuxa (np. Debian Sid działa poprawnie, występują tylko błędy związane z brakiem bibliotek, ale to 2 linijki do dopisania na cały projekt).

Jeżeli zależy ci wyłącznie na samych binarkach to możesz pobrać je z [pkgs.org](https://pkgs.org/download/colobot) lub dodać inne repozytoria do swojego managera pakietów.

**Comment by magic7003** [comment]
Bardzo dziękuję za odpowiedź .  Spóźnione co prawda .  Patrzyłem na GitHub a nie spodziewałem się odpowiedzi na pocztę elektroniczną . 
            
          
     
      
       
        Dnia 25 lipca 2021 20:30 ThePPK ***@***.***&gt; napisał(a):
       
    
       
         
 Jeżeli dobrze rozumiem to biblioteka SDL2 nie została załadowana, a jej powód to: 
 /usr/lib/i386-linux-gnu/libSDL2-2.0.so.0: error adding symbols: DSO missing from command line
 
 Na StackOverflow jest dyskusja na temat &#34;DSO missing from command line&#34;  stackoverflow.com LINK  
Przyczyną problemu mogą być niespójne zależności w systemie.  
Czyli np. SDL2 wymaga pthread a system twierdzi, że nie, więc  ld (linker) go nie ładuje. 
 Jeżeli bardzo ci zależy, aby ręcznie skompilować program, powinieneś ręcznie sprawdzić zależności dla SDL2. Możesz również wirtualizować innego linuxa (np. Debian Sid działa poprawnie, występują tylko błędy związane z brakiem bibliotek, ale to 2 linijki do dopisania na cały projekt). 
 Jeżeli zależy ci wyłącznie na samych binarkach to możesz pobrać je z  pkgs.org pkgs.org  lub dodać inne repozytoria do swojego managera pakietów.

**Comment by nikoleee2222** [comment]
This information was very helpful to me too!
